subprojects/gradle/buildchecks/src/main/kotlin/com/avito/android/plugin/build_param_check/BuildParamCheckPlugin.kt

```diff
@@ -112,6 +112,18 @@ open class BuildParamCheckPlugin : Plugin<Project> {
                 dependsOn(task)
             }
         }
+        if (checks.hasInstance<Check.UniqueRClasses>()) {
+            check(project.pluginManager.hasPlugin("com.avito.android.impact")) {
+                "build check for unique R classes requires 'com.avito.android.impact' plugin"
+            }
+            val task = project.tasks.register<UniqueRClassesTask>("checkUniqueAndroidPackages") {
+                group = "verification"
+                description = "Verify unique R classes"
+            }
+            checkBuildEnvironment {
+                dependsOn(task)
+            }
```

Comment:
One additional consideration.
I don't rely on `android.namespacedRClass` setting because this issue is relevant to both states.